apps/backend/src/auth/auth.service.spec.ts

```diff
@@ -0,0 +1,244 @@
+import { BadRequestException, ForbiddenException } from '@nestjs/common';
+import { ConfigModule, ConfigService } from '@nestjs/config';
+import { JwtModule, JwtService } from '@nestjs/jwt';
+import { Test, TestingModule } from '@nestjs/testing';
+
+import { hash } from 'bcrypt';
+import * as argon from 'argon2';
+
+import { userMock } from '../../test/mocks/user';
+import { UserService } from '../user/user.service';
+import { AuthService } from './auth.service';
+
+const userServiceMock = {
+  create: jest.fn(),
+  findById: jest.fn(),
+  findByEmail: jest.fn(),
+  update: jest.fn(),
+};
+
+const configServiceMock = {
+  get: jest.fn((key) => {
+    const values = {
+      JWT_ACCESS_SECRET: 'jwt_access_token_super_secret_key',
+      JWT_REFRESH_SECRET: 'jwt_refresh_token_super_secret_key',
+    };
+    return values[key] || null;
+  }),
+};
+
+const SALT_ROUNDS = 10;
+
+describe('AuthService', () => {
+  let service: AuthService;
+  let jwtService: JwtService;
+  let configService: ConfigService;
+  let userService: UserService;
+
+  beforeEach(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      imports: [JwtModule, ConfigModule],
+      providers: [
+        AuthService,
+        JwtService,
+        {
+          provide: ConfigService,
+          useValue: configServiceMock,
+        },
+        {
+          provide: UserService,
+          useValue: userServiceMock,
+        },
+      ],
+    }).compile();
+
+    service = module.get<AuthService>(AuthService);
+    jwtService = module.get<JwtService>(JwtService);
+    configService = module.get<ConfigService>(ConfigService);
+    userService = module.get<UserService>(UserService);
+  });
```

Comment:
Not a big deal, that's fairly well. However, I am a big fan of taking a more functional approach and creating a setup function and get rid of your `beforeEach`, something similar to this:

Outside of describe:

```js
const setup = () => {
 const module = await Test.createTestingModule({ ... })

 service = module.get<AuthService>(AuthService);
 jwtService = module.get<JwtService>(JwtService);

 return {
  service,
  jwtService,
 }
}
```

Then, on each test:
```js
it('should be defined', () => {
 const { service } = setup();

 expect(service).toBeDefined();
});
```

This way you ensure you have a totally new variable (any other test did not use that). I know your `beforeEach` is performing this job but I really think that using the `setup` approach makes your test more explicit for other developers.

WDYT?